apis/python/setup.py

```diff
@@ -27,9 +30,65 @@
 import version  # noqa E402
 
 
+def get_libtiledbsoma_library_name():
+    """
+    :return: List of TileDB shared library names.
+    """
+    if os.name == "posix":
+        if sys.platform == "darwin":
+            return "libtiledbsoma.dylib"
+        else:
+            return "libtiledbsoma.so"
+    elif os.name == "nt":
+        return "tiledbsoma.dll"
+    else:
+        raise RuntimeError(f"Unsupported OS name {os.name}")
+
+
+def find_libtiledbsoma_full_path_on_linux(lib_name):
+    # https://stackoverflow.com/questions/35682600/get-absolute-path-of-shared-library-in-python
+    class LINKMAP(ctypes.Structure):
+        _fields_ = [("l_addr", ctypes.c_void_p), ("l_name", ctypes.c_char_p)]
+
+    libdl = ctypes.CDLL(lib_name)
+    dlinfo = libdl.dlinfo
+    dlinfo.argtypes = ctypes.c_void_p, ctypes.c_int, ctypes.c_void_p
+    dlinfo.restype = ctypes.c_int
+
+    libc = ctypes.CDLL(lib_name)
+    lmptr = ctypes.c_void_p()
+    dlinfo(libc._handle, 2, ctypes.byref(lmptr))
+
+    return ctypes.cast(lmptr, ctypes.POINTER(LINKMAP)).contents.l_name.decode()
+
+
+def libtiledbsoma_exists():
+    """
+    Checks the given list of paths and returns true if any contain the TileDB library.
+    :return: The path to the TileDB library, or None.
```

Comment:
`The path to the TileDB library, or None.` -> `The path to the TileDB-SOMA library, or None.`